pkg/name/errors.go

```diff
@@ -32,6 +35,6 @@ func NewErrBadName(fmtStr string, args ...interface{}) *ErrBadName {
 
 // IsErrBadName returns true if the given error is an ErrBadName.
 func IsErrBadName(err error) bool {
-	_, ok := err.(*ErrBadName)
-	return ok
+	var berr *ErrBadName
+	return errors.As(err, &berr)
```

Comment:
I believe this can just be `errors.Is`

Reply:
I tried a bunch of different ways and couldn't get it to work.

It turns out this method is never used outside of tests in `pkg/name`, so I just removed it and inlined it (with `errors.As` for now) into the two places in tests where it was used. In theory someone is using it today, since it was exported. If we want I can put it back and deprecate it.

I also removed `NewErrBadName` which isn't used outside of `pkg/name`, and should not be needed outside of `pkg/name`.